Invalid env schema when rendering istio/gateway:1.12.0 chart with helm

https://github.com/istio/istio/blob/d6d17328c4e5f3ae0085d7896c9559420c841b86/manifests/charts/gateway/templates/deployment.yaml#L73

When using an array of the same form as a kubernetes env spec (name, value):
```
- name: istio-proxy
...
  env:
  - name: 0
    value: map[name:ISTIO_META_UNPRIVILEGED_POD value:true]
...
```

When using a map with form `key: value` as per the json schema:
```
coalesce.go:163: warning: skipped value for env: Not a table.
Error: values don't meet the specifications of the schema(s) in the following chart(s):
gateway:
- env: Invalid type. Expected: array, given: object
```

One of these is wrong; i personally prefer the kubernetes array form

--- a/manifests/charts/gateway/values.schema.json
+++ b/manifests/charts/gateway/values.schema.json
@@ -39,7 +39,7 @@
       }
     },
     "env": {
-      "type": "array"
+      "type": "object"
     },
     "labels": {
       "type": "object"

--- a/manifests/charts/gateway/values.yaml
+++ b/manifests/charts/gateway/values.yaml
@@ -68,7 +68,7 @@ autoscaling:
   targetCPUUtilizationPercentage: 80
 
 # Pod environment variables
-env: []
+env: {}
 
 # Labels to apply to all resources
 labels: {}

